tests/unit/parsers/test.manifestjson.js

```diff
@@ -4924,6 +5024,29 @@ describe('ManifestJSONParser', () => {
         code: messages.IGNORED_APPLICATIONS_PROPERTY.code,
       });
     });
+
+    it('emits a deprecation warning when a MV2 extension uses applications and an empty browser_specific_settings', () => {
+      const linter = new Linter({ _: ['bar'] });
+
+      const manifestJSONParser = new ManifestJSONParser(
+        JSON.stringify({
+          manifest_version: 2,
+          name: 'some name',
+          version: '1',
+          applications: {},
+          browser_specific_settings: {},
+        }),
+        linter.collector
+      );
+
+      expect(manifestJSONParser.isValid).toEqual(true);
+      expect(linter.collector.errors).toEqual([]);
+
+      expect(linter.collector.warnings.length).toEqual(1);
+      assertHasMatchingError(linter.collector.warnings, {
+        code: messages.APPLICATIONS_DEPRECATED.code,
+      });
```

Comment:
Nit, we could consider similar tweaks to this test case as the ones from the other two comments in this review pass.